### ISSUE TITLE
SqlServer database reverse engineering is inconsistent with script generator for unique indexes on nullable columns

### ISSUE THREAD

**Issue by azabluda** [dotnet/efcore#6156]
### Steps to reproduce

The failing unit test can be fetched from https://github.com/azabluda/EntityFramework/commit/1acdcdf648e61cdcb77f56fc9de850c9ff2b26cd.
### The issue

Nullable columns become non-nullable after translation of the result of RevEng back to DDL if there is a unique index defined on such columns.

Thus

``` sql
CREATE TABLE [dbo].[Place] (
    [Id] int NOT NULL IDENTITY,
    [Name] int,
    CONSTRAINT [PK_Place] PRIMARY KEY ([Id])
);
CREATE UNIQUE INDEX [IX_Place_Name] ON [dbo].[Place] ([Name]) WHERE [Name] IS NOT NULL;
```

becomes

``` sql
CREATE TABLE [dbo].[Place] (
    [Id] int NOT NULL IDENTITY,
    [Name] int NOT NULL,                         <<< FAIL
    CONSTRAINT [PK_Place] PRIMARY KEY ([Id]),
    CONSTRAINT [IX_Place_Name] UNIQUE ([Name])   <<< STRANGE
);
CREATE UNIQUE INDEX [IX_Place_Name] ON [dbo].[Place] ([Name]) WHERE [Name] IS NOT NULL;
```

One could speculate on whether the original DDL is itself ill-formed, but this is exactly what CodeFirst generates from my entity model.

This issue is possibly related to #3599.
### Further technical details

EF Core version: 1.0.0
Operating system: 
Visual Studio version: VS 2015

**Comment by rowanmiller** [comment]
Confirmed bug, reverse engineer marks the column as required when it is nullable in the database.

```c#
modelBuilder.Entity<Place>(entity =>
{
    entity.HasIndex(e => e.Name)
        .HasName("IX_Place_Name")
        .IsUnique();

    entity.Property(e => e.Name).IsRequired();
});
```

**Comment by lajones** [comment]
As of the fix for #7956 we no longer do that.
